core/templates/dev/head/domain/story_viewer/ReadOnlyStoryNodeObjectFactory.ts

```diff
@@ -17,72 +17,87 @@
  * story viewer.
  */
 
-angular.module('oppia').factory('ReadOnlyStoryNodeObjectFactory', [function() {
-  var StoryNode = function(
-      id, title, destinationNodeIds, prerequisiteSkillIds, acquiredSkillIds,
-      outline, outlineIsFinalized, explorationId, explorationSummary,
-      completed) {
-    this._id = id;
-    this._title = title;
-    this._destinationNodeIds = destinationNodeIds;
-    this._prerequisiteSkillIds = prerequisiteSkillIds;
-    this._acquiredSkillIds = acquiredSkillIds;
-    this._outline = outline;
-    this._outlineIsFinalized = outlineIsFinalized;
-    this._explorationId = explorationId;
-    this._explorationSummaryObject = explorationSummary;
-    this._completed = completed;
-  };
+import { downgradeInjectable } from '@angular/upgrade/static';
+import { Injectable } from '@angular/core';
 
-  // Instance methods
+export class StoryNode {
+  id: String;
+  title: String;
+  destinationNodeIds: Array<String>;
+  prerequisiteSkillIds: Array<String>;
+  acquiredSkillIds: Array<String>;
+  outline: String;
+  outlineIsFinalized: Boolean;
+  explorationId: String;
+  explorationSummary: Object;
+  completed: Boolean;
 
-  StoryNode.prototype.getId = function() {
-    return this._id;
-  };
+  constructor(id: String, title: String, destinationNodeIds: Array<String>,
+      prerequisiteSkillIds: Array<String>, acquiredSkillIds: Array<String>,
+      outline: String, outlineIsFinalized: Boolean, explorationId: String,
+      explorationSummary: Object, completed: Boolean) {
+    this.id = id;
+    this.title = title;
+    this.destinationNodeIds = destinationNodeIds;
+    this.prerequisiteSkillIds = prerequisiteSkillIds;
+    this.acquiredSkillIds = acquiredSkillIds;
+    this.outline = outline;
+    this.outlineIsFinalized = outlineIsFinalized;
+    this.explorationId = explorationId;
+    this.explorationSummary = explorationSummary;
+    this.completed = completed;
+  }
 
-  StoryNode.prototype.getTitle = function() {
-    return this._title;
-  };
+  getId(): String {
+    return this.id;
+  }
 
-  StoryNode.prototype.getExplorationId = function() {
-    return this._explorationId;
-  };
+  getTitle(): String {
+    return this.title;
+  }
 
-  StoryNode.prototype.isCompleted = function() {
-    return this._completed;
-  };
+  getExplorationId(): String {
+    return this.explorationId;
+  }
 
-  StoryNode.prototype.getExplorationSummaryObject = function() {
-    return this._explorationSummaryObject;
-  };
+  isCompleted(): Boolean {
+    return this.completed;
+  }
 
-  StoryNode.prototype.getOutline = function() {
-    return this._outline;
-  };
+  getExplorationSummaryObject(): Object {
```

Comment:
Also Object is not that clear, either create an interface for this, so that we know what are the keys in this object, or if that's not very clear just change that to any with a todo.